### PR TITLE
Support Multiple Pending Receives

### PR DESCRIPTION
**Pull request by nibanks**
## Description

Some internal refactoring in preparation to support multiple parallel receives to the app. No actual external behavior change yet. Just making sure no existing scenarios are broken.

## Testing

CI/CD

## Documentation

N/A

### CODE REVIEW

File: src/core/stream_recv.c [hunk 1]
```diff
@@ -1049,18 +1044,17 @@ QuicStreamReceiveComplete(
     _In_ uint64_t BufferLength
     )
 {
-    if (!Stream->Flags.ReceiveCallPending) {
+    if (!Stream->RecvCallPendingCount) {
         return FALSE;
     }
+    Stream->RecvCallPendingCount--;
```
Comment (nibanks) [inline src/core/stream_recv.c:1050]: This isn't going to work. With multiple receives, we want to support completing multiple with a single call too.

File: src/core/stream_recv.c [hunk 2]
```diff
@@ -1049,46 +1021,50 @@ QuicStreamReceiveComplete(
...
 
+    CXPLAT_TEL_ASSERTMSG(
+        BufferLength <= Stream->RecvPendingLength,
+        "App overflowed read buffer!");
+
     //
     // Reclaim any buffer space comsumed by the app.
     //
-    if (Stream->RecvPendingLength == 0 ||
+    if (!Stream->RecvPendingLength ||
```
Comment (csujedihy) [inline src/core/stream_recv.c:1044]: Why changing it to `!Stream->RecvPendingLength`? The implicit conversion from an integer to a boolean makes the code less readable.

File: src/core/stream.c [hunk 3]
```diff
@@ -195,10 +201,6 @@ QuicStreamFree(
     CxPlatDispatchLockUninitialize(&Stream->ApiSendRequestLock);
     CxPlatRefUninitialize(&Stream->RefCount);
 
-    if (Stream->ReceiveCompleteOperation) {
```
Comment (csujedihy) [inline src/core/stream.c:198]: where do we free this now?

File: src/core/stream_recv.c (hunk 2 above)
Comment (ProjectsByJackHe) [reply src/core/stream_recv.c:1044]: Maybe `Stream->RecvPendingLength == 0`?

File: src/core/stream.c (hunk 3 above)
Comment (nibanks) [reply src/core/stream.c:198]: It's allocated with the Stream itself, so when the stream is freed.

File: src/core/stream_recv.c (hunk 2 above)
Comment (ProjectsByJackHe) [reply src/core/stream_recv.c:1044]: +1 on this

Comment (nibanks) [reply src/core/stream_recv.c:1044]: I changed all the usages to this new model. What do folks prefer, the old version or the new?

**Review by csujedihy — APPROVED** [review]

**Pull request merged**